### PR TITLE
Add indent and sort keys support for JSON.print

### PR DESCRIPTION
**Pull request by Geequlim**
We can export the same json text content with same dictionary.
The VCS will be happy for that :)

### CODE REVIEW

File: core/io/json.cpp [hunk 1]
```diff
@@ -43,7 +43,25 @@ const char *JSON::tk_name[TK_MAX] = {
...
+		for (int i = 0; i < p_size; i++)
+			indent_text += p_indent;
+	}
+	return indent_text;
+}
+
+String JSON::_print_var(const Variant &p_var, const String& p_indent, int p_cur_indent, bool p_sort_keys) {
+
+	String colon = ":";
+	String end_statetment = "";
```
Comment (akien-mga) [inline core/io/json.cpp]: Typo in that variable name (state**t**ment).

File: core/io/json.cpp [hunk 2]
```diff
@@ -57,41 +75,50 @@ String JSON::_print_var(const Variant &p_var) {
...
 		};
 		case Variant::DICTIONARY: {
 
 			String s = "{";
+			s += end_statetment;
 			Dictionary d = p_var;
 			List<Variant> keys;
 			d.get_key_list(&keys);
 
+			if(p_sort_keys)
```
Comment (akien-mga) [inline core/io/json.cpp]: Style issue (space before `(`).

**Comment by Geequlim** [comment]
@akien-mga  Fixed and amend commited

**Comment by reduz** [comment]
Indent welcome, but sort keys is probably uneeded now, given Dictionary keys are now sorted by default in insertion order.

**Comment by Geequlim** [comment]
Like `json.dumps` does in python.

**Pull request merged**